File: objective-c/Ice/hello/HelloI.m
// **********************************************************************
//
// Copyright (c) 2003-2016 ZeroC, Inc. All rights reserved.
//
// **********************************************************************

#import <objc/Ice.h>
#import <HelloI.h>

#import <Foundation/NSThread.h>

#import <stdio.h>

@implementation HelloI
-(void) sayHello:(int)delay current:(ICECurrent*)c
{
    if(delay != 0)
    {
        [NSThread sleepForTimeInterval:delay / 1000.f];
    }
    printf("Hello World!\n");
    fflush(stdout);
}

-(void) shutdown:(ICECurrent*)c
{
    printf("Shutting down...\n");
    fflush(stdout);
    [[c.adapter getCommunicator] shutdown];
}

@end

File: cpp/IceGrid/customLocator/HelloI.cpp
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

#include <HelloI.h>
#include <Ice/Ice.h>

using namespace std;

HelloI::HelloI(string name) : _name(std::move(name)) {}

void
HelloI::sayHello(const Ice::Current&)
{
    cout << _name << " says Hello World!" << endl;
}

void
HelloI::shutdown(const Ice::Current& current)
{
    cout << _name << " shutting down..." << endl;
    current.adapter->getCommunicator()->shutdown();
}
